### ISSUE TITLE
Bug: problem of mounting the OPNsense kernel via netboot

### ISSUE THREAD

**Issue by EnikeevIlya** [opnsense/core#6553]
**Important notices**

Before you add a new report, we ask you kindly to acknowledge the following:

- [*] I have read the contributing guide lines at https://github.com/opnsense/core/blob/master/CONTRIBUTING.md
- [*] I am convinced that my issue is new after having checked both open and closed issues at https://github.com/opnsense/core/issues?q=is%3Aissue

**Describe the bug**

I am facing a problem when mounting the OPNsense kernel via network boot (https://github.com/netbootxyz )

**To Reproduce**

1. I have a USB drive with a netboot.xyz image.
2. I am running the netboot.xyz application from a USB drive on my computer (booting from BIOS)
3. On the IPXE command line (or in the application netboot) I upload an OPNsense image via "sanboot" (previously unpacked from the archive."bz2") from a remote server and I'm trying to install it on my computer.
It looks like this:
`sanboot http://my-local-server/OPNsense_dvd_image.iso`
4. I see the OPNsense boot menu, then in the "autoboot" process the necessary modules are loaded, the hardware is polled, and then I get an error message `Mounting from: cd9660:/dev/iso9660/OPNSENSE_INSTALL failed 19.`

**Expected behavior**

Successful installation of OPNsense. For example, there is no such problem with installing FreeBSD, although the installation steps are the same, only the FreeBSD image and not OPNsense.

**Describe alternatives you considered**

To begin with, to check the correctness of the OPNsense DVD image that I have in stock, I burned it to a USB drive and started the installation directly on my computer (without using the netboot) and the download and installation was successful.
Next, I tried different images (dvd, vga, serial) and their versions taken from https://opnsense.org/download/, but in all cases the installation broke down in the same place and I got a similar error.
The search for articles with such an error did not yield results, since it describes a similar problem with installing OPNsense from a USB drive directly to a computer. However, I have tried almost all the options that have helped people in these articles: starting with changing the settings in the BIOS, ending with changing variables (like "vfs.root.mountfrom") at the moment of the command line of the OPNsense loader.
Next, I installed FreeBSD 13.1 on test hardware, there I assembled an OPNsense image according to the instructions - but installing this image via netboot also did not give results.
Next, before assembling the OPNsense image in the file `/opnsense/tools/build/dvd.sh` before `"Building dvd image..."` I "planted" the loader instead of the default `cdboot` root from FreeBSD `mfsroot.gz `. And collected the resulting OPNsense image: as a result, despite the fact that the OPNsense loader, I have successfully installed FreeBSD. Hence, I would venture to assume that OPNsense has a problem with the roots, for example, with `cdboot`, which does not allow it to be installed seamlessly via a netboot download.

**Screenshots**

![1](https://github.com/opnsense/core/assets/95696127/03a624a0-986b-4bd3-9b4c-9c98f2e9b162)
![2](https://github.com/opnsense/core/assets/95696127/7a93e6c8-b842-475a-b9ca-e5633f4dda8c)

**Relevant log files**

Unfortunately, I cannot provide the full logs of the OPNsense installation, as the installation fails. Nevertheless, I have the opportunity to "rip out" screenshots from the video where I filmed the OPNsense installation process, in case you need it, and also provide any additional information upon your request.

**Additional context**

Let me clarify that this problem occurs not only with the "dvd" image, but also with "vga" and "serial" and any available version of them.

**Environment**

Software version used and hardware type if relevant, e.g.:

OPNsense 23.1 (amd64, OpenSSL).
Intel® Celeron® J4125; Quad Core 2.0 Ghz

**Comment by AdSchellevis** [comment]
Marking this community support in case someone wants to figure this out (we don't officially support netboot), personally I wouldn't expect this would work out of the box in a pxe environment, some earlier references on the FreeBSD forum suggest the same 
https://forums.freebsd.org/threads/install-disk-via-pxe.80319/#post-510824

**Comment by EnikeevIlya** [comment]
> Marking this community support in case someone wants to figure this out (we don't officially support netboot), personally I wouldn't expect this would work out of the box in a pxe environment, some earlier references on the FreeBSD forum suggest the same https://forums.freebsd.org/threads/install-disk-via-pxe.80319/#post-510824

Hello. So, I created an iso image of OPNsense, when installing it passes the above error. But I have a question.:
Below I provide screenshots of the download of this OPNsense image, compiled from the sources in this repository. Is this what an OPNsense installation should look like? I strongly doubt it, because installing images with https://opnsense.org/download / looks completely different, but here it looks more like a FreeBSD installation.

![1](https://github.com/opnsense/core/assets/95696127/8ac3abf9-882d-4fdd-b194-6b82c902c3f2)
![2](https://github.com/opnsense/core/assets/95696127/69201e82-9012-44ff-a00f-679dd8be6b9e)
![3](https://github.com/opnsense/core/assets/95696127/13e4a828-170e-468c-932a-e9d7713dec92)
![4](https://github.com/opnsense/core/assets/95696127/03c40487-8acd-4ace-ba1f-3612837b1dfe)
![5](https://github.com/opnsense/core/assets/95696127/85648d7e-c73a-47bf-974c-45a114ff3231)

Тем более, после установки я наблюдаю следующее:

```
root@:~# uname -a
FreeBSD 13.2-RELEASE-p3 FreeBSD 13.2-p3 stable/23.7-n254808-08aa869336f SMP amd64
```

**Comment by AdSchellevis** [comment]
I'm not sure what you mean, but we don't officially support custom builds, only offer the tools to do so (https://github.com/opnsense/tools)

**Comment by EnikeevIlya** [comment]
> I'm not sure what you mean, but we don't officially support custom builds, only offer the tools to do so (https://github.com/opnsense/tools)

@AdSchellevis thanks for your reply. I'm sorry if I didn't make myself clear.
I mean, I used the tool that you provide (https://github.com/opnsense/tools), and now the OPNsense installation process looks like it is shown in the screenshots that I posted. Is this how this process should look visually? It is very similar to the FreeBSD installation process.
